src/main/java/tutoraid/logic/commands/EditStudentCommand.java

```diff
@@ -0,0 +1,212 @@
+    /**
+     * Stores the details to edit the person with. Each non-empty field value will replace the
+     * corresponding field value of the person.
+     */
+    public static class EditStudentDescriptor {
+        private StudentName studentName;
+        private Phone studentPhone;
+        private ParentName parentName;
+        private Phone parentPhone;
+
+        // Data fields
+        private Progress progress;
+        private PaymentStatus paymentStatus;
+
+        public EditStudentDescriptor() {
+        }
+
+        /**
+         * Copy constructor.
+         */
+        public EditStudentDescriptor(EditStudentDescriptor toCopy) {
+            setStudentName(toCopy.studentName);
+            setStudentPhone(toCopy.studentPhone);
+            setParentName(toCopy.parentName);
+            setParentPhone(toCopy.parentPhone);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(studentName, studentPhone, parentName, parentPhone);
+        }
+
+        public void setStudentName(StudentName studentName) {
+            this.studentName = studentName;
+        }
+
+        public Optional<StudentName> getStudentName() {
+            return Optional.ofNullable(studentName);
+        }
+
+        public void setStudentPhone(Phone studentPhone) {
+            this.studentPhone = studentPhone;
+        }
+
+        public Optional<Phone> getStudentPhone() {
+            return Optional.ofNullable(studentPhone);
+        }
+
+        public void setParentName(ParentName parentName) {
+            this.parentName = parentName;
+        }
+
+        public Optional<ParentName> getParentName() {
+            return Optional.ofNullable(parentName);
+        }
+
+        public void setParentPhone(Phone parentPhone) {
+            this.parentPhone = parentPhone;
+        }
+
+        public Optional<Phone> getParentPhone() {
+            return Optional.ofNullable(parentPhone);
+        }
+
+        /**
+         * Returns true if both descriptors represent students that have the same identity and data fields.
+         * This defines a stronger notion of equality between two students.
+         */
+        @Override
+        public boolean equals(Object other) {
+            if (other == this) {
+                return true;
+            }
+
+            if (!(other instanceof EditStudentDescriptor)) {
+                return false;
+            }
+
+            EditStudentDescriptor e = (EditStudentDescriptor) other;
+            EditStudentDescriptor otherDescriptor = (EditStudentDescriptor) other;
+            return otherDescriptor.getStudentName().equals(getStudentName())
+                && otherDescriptor.getStudentPhone().equals(getStudentPhone())
+                && otherDescriptor.getParentName().equals(getParentName())
+                && otherDescriptor.getParentPhone().equals(getParentPhone());
```

Comment:
I was wondering if we need to compare the payment status and progress also. It seems we don't need to compare them because those are not edited by this EditStudentCommand?